code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -144,8 +144,10 @@
 	if(!length(result_atoms))
 		return FALSE
 
-	for(var/result in result_atoms)
-		new result(loc)
+	for(var/atom/result in result_atoms)
+		result = new result(loc)
+		if(isitem(result))
+			ADD_TRAIT(result, TRAIT_CONTRABAND, TRAIT_GENERIC)
```

Comment:
This should definitely be breaking unit test, you typed the var to `atom` but there's no atoms in the list
```suggestion
	for(var/result in result_atoms)
		var/atom/result_item = new result(loc)
		if(isitem(result_item ))
			ADD_TRAIT(result_item, TRAIT_CONTRABAND, TRAIT_GENERIC)
```